assets/js/components/devices/import/ChooseImportType.jsx

```diff
@@ -1,108 +1,151 @@
-      <Text style={{width: '100%', textAlign: 'center', margin: '0px 40px 10px'}}>
-        <b>Sorry. The file you supplied doesn’t appear to be formatted correctly.</b>
-        <br/>
-        Please ensure it is a csv file, with correctly formatted headers.
+const FormatPopover = () => (
+  <Popover
+    trigger="click"
+    content={
+      <Text style={{ display: "table", whiteSpace: "normal", width: 200 }}>
+        The fields <b>DevEUI, AppEUI, AppKey,</b> and <b>Name</b> are all
+        required. Optionally you can include a <b>LabelID</b> column to add
+        labels to imported devices.
       </Text>
-      <Popover
-        onVisibleChange={() => setOpen(!open)}
-        visible={open}
-        trigger="click"
-        content={
-          <Text style={{display: 'table', whiteSpace: 'normal', width: 200}}>
-            The fields <b>DevEUI, AppEUI, AppKey,</b> and <b>Name</b> are all required. Optionally you can include a <b>LabelID</b> column to add labels to imported devices.
-          </Text>
-        }
-      >
-        <Text style={{width: '100%', textAlign: 'center', marginBottom: 20, color: 'deepskyblue', textDecoration: 'underline'}}>How do I format my .csv?</Text>
-      </Popover>
-      <DragAndDrop fileSelected={(file) => {
-        let fileReader = new FileReader();
-        fileReader.onloadend = () => {
-          scanGenericDevices(fileReader.result, (type) => onImportSelect(type));
-        }
-        fileReader.readAsText(file);
-      }}>
-        <Text style={{textAlign: 'center', margin: '30px 80px', color: blueForDeviceStatsLarge}}>
-          Drag .csv file here or click to choose file
+    }
+  >
+    <a className="help-link" style={{ marginBottom: 20, color: "deepskyblue" }}>
+      How do I format my .csv?
+    </a>
+  </Popover>
+);
+
+const ChooseImportType = ({
+  onImportSelect,
+  scanGenericDevices,
+  genericImportScanFailed,
+  deviceImports,
+}) => {
+  if (genericImportScanFailed)
+    return (
+      <Fragment>
+        <img
+          src={Warning}
+          style={{ marginBottom: 10, height: 50, objectFit: "cover" }}
+        />
+        <Title style={{ fontSize: 26, width: "100%", textAlign: "center" }}>
+          Invalid Filetype
+        </Title>
+        <Text
+          style={{
+            width: "100%",
+            textAlign: "center",
+            margin: "0px 40px 10px",
+          }}
+        >
+          <b>
+            Sorry. The file you supplied doesn’t appear to be formatted
+            correctly.
+          </b>
+          <br />
+          Please ensure it is a CSV file, with correctly formatted headers.
         </Text>
-      </DragAndDrop>
-    </Fragment>
-  )
+        <FormatPopover />
+        <DragAndDrop
+          fileSelected={(file) => {
+            let fileReader = new FileReader();
+            fileReader.onloadend = () => {
+              scanGenericDevices(fileReader.result, (type) => {
+                if (type) onImportSelect(type);
```

Comment:
type is `""` when the import fails (i.e. for invalid file) so it should not continue.